### PR TITLE
fix #839

### LINKED ISSUE #839: Read IDs of ensemble_prediction from hdf5 as list/tuple
```
Traceback (most recent call last):
  File "/home/simon/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/201.7846.77/plugins/python/helpers/pydev/pydevd.py", line 1438, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/home/simon/.local/share/JetBrains/Toolbox/apps/PyCharm-P/ch-0/201.7846.77/plugins/python/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/home/simon/Documents/covid-ethiopia/variants/code/models/v0/sample_model_v0.py", line 183, in <module>
    axes = visualize.sampling_prediction_trajectories(
  File "/home/simon/git/pyPESTO/pypesto/visualize/sampling.py", line 685, in sampling_prediction_trajectories
    _plot_trajectories_by_output(
  File "/home/simon/git/pyPESTO/pypesto/visualize/sampling.py", line 302, in _plot_trajectories_by_output
    t_median, y_median = _get_statistic_data(
  File "/home/simon/git/pyPESTO/pypesto/visualize/sampling.py", line 163, in _get_statistic_data
    condition_index = summary[statistic].condition_ids.index(condition_id)
AttributeError: 'numpy.ndarray' object has no attribute 'index'
```

### PR DESCRIPTION
**Pull request by MerktSimon**
fixes #839 
Reading ensemble_predictions from hdf5 files can lead to some issues since the read IDs are saved as numpy.arrays instead of lists, which is used in the original creation of the ensemble_prediction. This will be fixed with this PR.

### CODE REVIEW

File: pypesto/ensemble/util.py [hunk 1]
```diff
@@ -316,9 +316,7 @@ def read_ensemble_prediction_from_h5(
                 bounds[key] = np.array(bounds[key])
                 continue
             x_names = decode_array(f[f'{key}/{X_NAMES}'][()])
-            condition_ids = np.array(
-                decode_array(f[f'{key}/condition_ids'][()])
-            )
+            condition_ids = decode_array(f[f'{key}/condition_ids'][()])
```
Comment (dilpath) [inline pypesto/ensemble/util.py:319]: Does this do anything? Fine to leave as-is, just curious.

```suggestion
            condition_ids = decode_array(f[f'{key}/condition_ids'])
```

Comment (dilpath) [reply pypesto/ensemble/util.py:319]: Also elsewhere there are `[:]` that probably don't do anything? But fine to merge I think

**Review by dilpath — APPROVED** [review]
:+1:

File: pypesto/ensemble/util.py (hunk 1 above)
Comment (MerktSimon) [reply pypesto/ensemble/util.py]: Does not work without. I think it does return array instead of h5py dataset.

**Pull request merged**